### DISCUSSION TITLE
Managing a graph-per-call state

### DISCUSSION THREAD

**Discussion by yuval-airis** [dagster-io/dagster#18358] [Q&A] [answered]
I have a use case where I get calls through some endpoint, and then calculate assets A,B,C for each of these calls, and save the combined results of A,B,C somewhere.

I have several open issues:
1) First triggering, suppose I have some non-dagster thread getting triggers and starting new dagster flows. How may I do that? Would "materialize" with a suitable config be a good choice?
How may I retrieve / monitor the results of the run afterwards?

2) Secondly, suppose that the graph run would be dependent on inputs from the trigger. (e.g. the name of an input file I got), should it be passed as a config to the relevant op/s?
 
3) If the graph does fail at a later point for some of the inputs, how may I notice that and rerun the remaining steps? Can I see which runs failed and in which ops/assets in the UI somehow?

**Comment by smackesey** [comment, marked as answer]
>First triggering, suppose I have some non-dagster thread getting triggers and starting new dagster flows. How may I do that? Would "materialize" with a suitable config be a good choice?
How may I retrieve / monitor the results of the run afterwards?

`materialize` is more of a test utility. While technically you could use it, you'd need a reference to your persistent Dagster instance. I'd recommend running a [sensor](https://docs.dagster.io/concepts/partitions-schedules-sensors/sensors) instead that monitors some queue, and have your incoming requests write to that queue. A running sensor can check the queue every 30 seconds.

>Secondly, suppose that the graph run would be dependent on inputs from the trigger. (e.g. the name of an input file I got), should it be passed as a config to the relevant op/s?

Yes. You can do this when assembling the `RunRequest` in your sensor.

>If the graph does fail at a later point for some of the inputs, how may I notice that and rerun the remaining steps? Can I see which runs failed and in which ops/assets in the UI somehow?

See https://docs.dagster.io/guides/dagster/re-execution
